### PR TITLE
*: add a doc and related description for the Optimizer Fix Controls

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

As the title says, add a page add related description for the Optimizer Fix Control.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/14187
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: optimizer-fix-controls.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+---
+
+# Optimizer Fix Controls
+
+The TiDB optimizer has lots of details in its behavior. As TiDB evolves, these details are changing. Usually, these changes are improvements of the optimizer. But sometimes, they would possibly cause unexpected results like execution plan regression:
+
+- For some implementation details, different behaviors are more suitable to different scenarios.
+- Sometimes, the relationship between changes in the behavior details and their consequences is very complicated. Even if it's an improvement in a certain behavior, it might cause execution plan regression as a whole.
+
+So we provided the [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-new-in-v710) system variable to control the behavior of the optimizer in a more fine-grained way.
```
Comment (hfxsd) [inline optimizer-fix-controls.md:14]:
```suggestion
Starting from v7.1.0, TiDB provides the [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-new-in-v710) system variable to control the behavior of the optimizer in a more fine-grained way.
```

File: optimizer-fix-controls.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+A Fix is usually denoted by a number, which usually corresponds to a GitHub Issue, in which there will be descriptions for the technical details. For example, Fix `44262` corresponds to [Issue 44262](https://github.com/pingcap/tidb/issues/44262).
```
Comment (hfxsd) [inline optimizer-fix-controls.md:16]:
```suggestion
A fix is usually denoted by a number, which usually corresponds to a GitHub Issue. There are descriptions about the technical details in the GitHub issue. For example, Fix `44262` corresponds to [Issue 44262](https://github.com/pingcap/tidb/issues/44262).
```

File: optimizer-fix-controls.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+This variable supports multiple Fixes, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` is the Fix number. For example:
```
Comment (hfxsd) [inline optimizer-fix-controls.md:18]:
```suggestion
This variable supports multiple fixes, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` is the fix number. For example:
```

File: optimizer-fix-controls.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+- Default value: `OFF`
+- Possible values: `ON`, `OFF`
+- This variable controls whether to allow the use of [Dynamic pruning mode](/partitioned-table.md#dynamic-pruning-mode) to access the partitioned table when the [GlobalStats](/statistics.md#collect-statistics-of-partitioned-tables-in-dynamic-pruning-mode) are missing.
+
+### [`44389`](https://github.com/pingcap/tidb/issues/44389) <span class="version-mark">New in v7.1.1</span>
+
+- Default value: `OFF`
+- Possible values: `ON`, `OFF`
+- For filters like `c = 10 and (a = 'xx' or (a = 'kk' and b = 1)`, whether to try to build more complete scan ranges for `IndexRangeScan`.
```
Comment (hfxsd) [inline optimizer-fix-controls.md:36]:
```suggestion
- For filters such as `c = 10 and (a = 'xx' or (a = 'kk' and b = 1)`, this variable controls whether to try to build more comprehensive scan ranges for `IndexRangeScan`.
```

File: system-variables.md [hunk 5]
```diff
@@ -3383,7 +3383,7 @@ mysql> desc select count(distinct a) from test.t;
 - Default value: `""`
 - This variable is used to control some internal behaviors of the optimizer.
 - The optimizer's behavior might vary depending on user scenarios or SQL statements. This variable provides a more fine-grained control over the optimizer and helps to prevent performance regression after upgrading caused by behavior changes in the optimizer.
-- This variable supports multiple control items, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` represents the control number, corresponding to the link `https://github.com/pingcap/tidb/issues/<#issue>` that describes the control item, and `<valueN>` is the target value set for the control item.
+- Reference [Optimizer Fix Controls](/optimizer-fix-controls.md) for a more detailed introduction.
```
Comment (hfxsd) [inline system-variables.md:3386]:
```suggestion
- For a more detailed introduction, see [Optimizer Fix Controls](/optimizer-fix-controls.md).
```

File: optimizer-fix-controls.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+A Fix is usually denoted by a number, which usually corresponds to a GitHub Issue, in which there will be descriptions for the technical details. For example, Fix `44262` corresponds to [Issue 44262](https://github.com/pingcap/tidb/issues/44262).
+
+This variable supports multiple Fixes, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` is the Fix number. For example:
+
+```sql
+SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
+```
+
+## Optimizer Fix Controls Reference
```
Comment (hfxsd) [inline optimizer-fix-controls.md:24]:
```suggestion
## Optimizer Fix Controls reference
```

File: optimizer-fix-controls.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+---
+title: Optimizer Fix Controls
+summary: Learn about the Optimizer Fix Controls feature and how to use `tidb_opt_fix_control` to control the TiDB optimizer in a more fine-grained way.
+---
+
+# Optimizer Fix Controls
+
+The TiDB optimizer has lots of details in its behavior. As TiDB evolves, these details are changing. Usually, these changes are improvements of the optimizer. But sometimes, they would possibly cause unexpected results, such as:
+
+- For some implementation details, different behaviors are more suitable to different scenarios. Some changes that bring improvements for most scenarios may also cause regressions for a few of other scenarios.
```
Comment (hfxsd) [inline optimizer-fix-controls.md:10]:
```suggestion
- For some implementation details, different behaviors are more suitable to different scenarios. Some changes that bring improvements for most scenarios might also cause regressions for a few of other scenarios.
```

File: optimizer-fix-controls.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
+---
+title: Optimizer Fix Controls
+summary: Learn about the Optimizer Fix Controls feature and how to use `tidb_opt_fix_control` to control the TiDB optimizer in a more fine-grained way.
+---
+
+# Optimizer Fix Controls
+
+The TiDB optimizer has lots of details in its behavior. As TiDB evolves, these details are changing. Usually, these changes are improvements of the optimizer. But sometimes, they would possibly cause unexpected results, such as:
```
Comment (hfxsd) [inline optimizer-fix-controls.md:8]:
```suggestion
The TiDB optimizer has lots of details in its behavior. As TiDB evolves, these details are changing. Usually, these changes are improvements of the optimizer. But sometimes, they might cause unexpected results, such as:
```

File: optimizer-fix-controls.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+- For some implementation details, some behaviors are more suitable for certain scenarios. Some changes that bring improvements for most scenarios might cause regressions for other scenarios.
```
Comment (hfxsd) [inline optimizer-fix-controls.md:10]:
```suggestion
- For some implementation details, some behaviors are more suitable for certain scenarios. Changes that bring improvements for most scenarios might cause regressions for other scenarios.
```

File: optimizer-fix-controls.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+- Sometimes, the relationship between changes in the behavior details and their consequences is very complicated. Even if it is an improvement in a certain behavior, it might cause execution plan regression as a whole.
+
```
Comment (hfxsd) [inline optimizer-fix-controls.md:12]:
```suggestion

## `tidb_opt_fix_control`

```

**Review by hfxsd — APPROVED** [review]

File: optimizer-fix-controls.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
+---
+title: Optimizer Fix Controls
+summary: Learn about the Optimizer Fix Controls feature and how to use `tidb_opt_fix_control` to control the TiDB optimizer in a more fine-grained way.
+---
+
+# Optimizer Fix Controls
+
+The TiDB optimizer has lots of details in its behavior. As TiDB evolves, these details are changing. Usually, these changes are improvements of the optimizer. But sometimes, they might cause unexpected results, such as:
```
Comment (hfxsd) [inline optimizer-fix-controls.md:8]:
```suggestion
As the product evolves iteratively, the behavior of the TiDB optimizer changes, which in turn generates more reasonable execution plans. However, in some specific scenarios, the new behavior may lead to unintended results. For example:
```

File: optimizer-fix-controls.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+- For some implementation details, some behaviors are more suitable for certain scenarios. Changes that bring improvements for some scenarios might cause regressions for others.
```
Comment (hfxsd) [inline optimizer-fix-controls.md:10]:
```suggestion
- The effect of some behaviors is related to a specific scenario. Changes that bring improvements for some scenarios might cause regressions for others.
```

File: optimizer-fix-controls.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
+---
+title: Optimizer Fix Controls
+summary: Learn about the Optimizer Fix Controls feature and how to use `tidb_opt_fix_control` to control the TiDB optimizer in a more fine-grained way.
+---
+
+# Optimizer Fix Controls
+
+As the product evolves iteratively, the behavior of the TiDB optimizer changes, which in turn generates more reasonable execution plans. However, in some specific scenarios, the new behavior may lead to unintended results. For example:
```
Comment (songrijie) [inline optimizer-fix-controls.md:8]:
```suggestion
As the product evolves iteratively, the behavior of the TiDB optimizer changes, which in turn generates more reasonable execution plans. However, in some particular scenarios, the new behavior may lead to unexpected results. For example:
```

File: optimizer-fix-controls.md [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+
+- The effect of some behaviors relies on a specific scenario. Changes that bring improvements for some scenarios might cause regressions for others.
+- Sometimes, the relationship between changes in the behavior details and their consequences is very complicated. An improvement in a certain behavior might cause execution plan regression as a whole.
```
Comment (songrijie) [inline optimizer-fix-controls.md:11]:
```suggestion
- Sometimes, the relationship between changes in the behavior details and their consequences is very complicated. An improvement in a certain behavior might cause overall regression.
```

File: optimizer-fix-controls.md [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+Therefore, TiDB provides the Optimizer Fix Controls feature that allows you to control the details of TiDB optimizer behaviors by setting a series of fixes. This document describes the Optimizer Fix Controls feature and how to use them, and lists all the fixes that TiDB currently supports.
```
Comment (songrijie) [inline optimizer-fix-controls.md:13]:
```suggestion
Therefore, TiDB provides the Optimizer Fix Controls feature that allows you to make fine-grained control of TiDB optimizer behaviors by setting values for a group of fixes. This document describes the Optimizer Fix Controls feature and how to use them, and lists all the fixes that TiDB currently supports for Fix Controls.
```

File: optimizer-fix-controls.md [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+
+## Introduction to `tidb_opt_fix_control`
+
+Starting from v7.1.0, TiDB provides the [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-new-in-v710) system variable to control the behavior of the optimizer in a more fine-grained way.
+
+A fix is a control item used to adjust the behavior in the TiDB optimizer at one point. It is denoted by a number that corresponds to a GitHub Issue. The GitHub issue contains the technical details. For example, Fix `44262` corresponds to [Issue 44262](https://github.com/pingcap/tidb/issues/44262).
```
Comment (songrijie) [inline optimizer-fix-controls.md:19]:
```suggestion
Each fix is a control item used to adjust the behavior in the TiDB optimizer for one particular purpose. It is denoted by a number that corresponds to a GitHub Issue, that contains the technical details of the behavior change. For example, for fix `44262`, you can review what it controls in [Issue 44262](https://github.com/pingcap/tidb/issues/44262).
```

File: optimizer-fix-controls.md [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+
+This variable supports multiple fixes, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` is the fix number. For example:
```
Comment (songrijie) [inline optimizer-fix-controls.md:21]:
```suggestion
Variable [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-new-in-v710) accepts multiple fixes as one value, separated by commas (`,`). The format is `"<#issue1>:<value1>,<#issue2>:<value2>,...,<#issueN>:<valueN>"`, where `<#issueN>` is the fix number. For example:
```

**Review by songrijie — APPROVED** [review]

**Comment by ran-huang** [comment]
/review 1. Evaluate the title, description, and diff of the PR. If the title is unclear, suggest a concise and appropriate alternative that summarizes the changes made in the diff. 2. Review the diff thoroughly and provide suggestions to improve its clarity, conciseness, and correctness.

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 18]
```diff
@@ -3377,13 +3377,29 @@ mysql> desc select count(distinct a) from test.t;
...
+
+<CustomContent platform="tidb-cloud">
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Type: String
+- Default value: `""`
+- This variable is used to control some internal behaviors of the optimizer.
+- The optimizer's behavior might vary depending on user scenarios or SQL statements. This variable provides a more fine-grained control over the optimizer and helps to prevent performance regression after upgrading caused by behavior changes in the optimizer.
+- For a more detailed introduction, see [Optimizer Fix Controls](https://docs.pingcap.com/tidb/stable/optimizer-fix-controls).
```
Comment (hfxsd) [inline system-variables.md:3400]:
```suggestion
- For a more detailed introduction, see [Optimizer Fix Controls](https://docs.pingcap.com/tidb/v7.2/optimizer-fix-controls).
```

**Pull request merged**